src/ik_evolution_2.cpp

```diff
@@ -266,12 +270,12 @@ template <int memetic> struct IKEvolution2 : IKBase
             auto& parent = population[0];
             auto& parent2 = population[1];
             double fmix = (child_index % 2 == 0) * 0.2;
-            double gradient_factor = child_index % 3;
+            double gradient_factor = static_cast<double>(child_index % 3);
 
             auto __attribute__((aligned(32)))* __restrict__ parent_genes = parent.genes.data();
             auto __attribute__((aligned(32)))* __restrict__ parent_gradients = parent.gradients.data();
 
-            auto __attribute__((aligned(32)))* __restrict__ parent2_genes = parent2.genes.data();
+            [[maybe_unused]] auto __attribute__((aligned(32)))* __restrict__ parent2_genes = parent2.genes.data();
```

Comment:
If this never gets used then just delete it.

Reply:
It's referenced in the `#pragma omp simd` just below it, but I suppose that it doesn't need to be aligned if it's not actually used later on. Will remove

Reply:
If we later use it, why did you add `[[maybe_unused]]`?

Reply:
I guess because gcc doesn't think that referencing a variable in a `pragma` is using it? It's not actually used anywhere else except to specify the alignment

Reply:
Removed in [e9c01ee](https://github.com/PickNikRobotics/bio_ik/pull/1/commits/e9c01ee4ef00ced5d79bb788ab01a067e93e922c)